File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let currkSum = 0;
    for (let i = 0; i < k; i++) {
        currkSum += nums[i];
    }

    let maxkSum = currkSum;
    for (let i = k; i < nums.length; i++) {
        currkSum += nums[i] - nums[i - k];
        maxkSum = Math.max(maxkSum, currkSum);
    }
    
    return maxkSum/k;
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int max = Integer.MIN_VALUE;
        int curr_sum = 0;
        
        for (int i = 0; i < k; i++) {
            curr_sum += nums[i];
        }
        max = Math.max(max, curr_sum);

        for (int i = k; i < nums.length; i++) {
            curr_sum = curr_sum + nums[i] - nums[i - k];
            max = Math.max(curr_sum, max);
        }

        return (double) max / k;

    }
}